### PR TITLE
Change checklist order on post edit meta box.

### PR DESCRIPTION
**Pull request by julianactrl**
Changes for checklist order on post edit meta box

### CODE REVIEW

File: modules/checklists/checklists.php [hunk 1]
```diff
@@ -640,6 +640,23 @@ public function display_meta_box($post)
...
+            }
+
+            $new_arr = array_intersect_key( $options, $requirement_rule_array );
+
+            $requirement_rule_array = array_merge(array_flip(array_keys( $new_arr) ), $requirement_rule_array);
+
+            foreach ($requirement_rule_array as $req_index) {
+                $new_requirements_array[$req_index] = $requirements[$req_index];
+            };
+
```
Comment (andergmartins) [inline modules/checklists/checklists.php:659]: I see a very similar code added to function `options_page_controller`, same file lines 877 to 893.

Please, avoid duplicated code. If you need the same code in multiple places it is required to create a new method encapsulating it, so you can call it on both places.

Please, could you refactor it to remove the duplication?

**Review by andergmartins — CHANGES_REQUESTED** [review]

File: modules/checklists/checklists.php (hunk 1 above)
Comment (andergmartins) [reply modules/checklists/checklists.php:659]: But the feature works well now 👍  I see the sorted list both on the settings and post edit page.

Comment (julianactrl) [reply modules/checklists/checklists.php]: @andergmartins  I removed the duplicated codes and pushed again.
Please check commits again.

Thanks.

Comment (andergmartins) [reply modules/checklists/checklists.php]: Thanks @julianactrl I confirm that works.

**Review by andergmartins — APPROVED** [review]

**Pull request merged**